src/main/java/components/Deadline.java

```diff
@@ -0,0 +1,40 @@
+package components;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents deadline tasks.
+ */
+public class Deadline extends Task {
+
+  protected String by;
+  protected LocalDate by2;
```

Comment:
Perhaps the variable name should be dateDueBy instead of by2?

Reply:
Agreed! More clarity to variable names would help with your code readability!